pandas/core/categorical.py

```diff
@@ -347,9 +334,14 @@ def __init__(self, values, categories=None, ordered=None, dtype=None,
                 raise NotImplementedError("> 1 ndim Categorical are not "
                                           "supported at this time")
 
-            if dtype.categories is None:
-                # we're inferring from values
-                dtype = CategoricalDtype(categories, ordered)
+            # we're inferring from values
+            dtype = CategoricalDtype(categories, dtype.ordered)
+
+        elif is_categorical_dtype(values):
```

Comment:
is this a repeat of the first if?? (this one actually looks good), wondering why it is hitting twice.

Reply:
it's not a repeat, as here the actual 'codes' construction happens (before it was to check the categories)